api/providers/tasks.py

```diff
@@ -522,24 +530,41 @@ def handle_registration_row(row, initiator, provider, schema, auto_approval=Fals
     draft.save()
     row.is_completed = True
     row.save()
-    logger.info('Draft registration created: [{}]'.format(row.draft_registration.id))
+    logger.info('Draft registration created: [{}]'.format(row.draft_registration._id))
+
+    # Register the draft since responses have been validated already
+    try:
+        registration = draft.register(auth, save=True)
+    except NodeStateError as e:
+        error = f'Fail to register draft: {repr(e)}'
+        raise RegistrationBulkCreationRowError(row.upload.id, row.id, row_title, row_external_id, error=error)
+    except Exception as e:
+        raise RegistrationBulkCreationRowError(row.upload.id, row.id, row_title, row_external_id, error=repr(e))
+    logger.info('Registration [{}] created from draft [{}]'.format(registration._id, row.draft_registration._id))
+
+    # Requires approval
+    try:
+        registration.require_approval(initiator)
+    except NodeStateError as e:
+        error = f'Fail to require approval: {repr(e)}'
+        raise RegistrationBulkCreationRowError(row.upload.id, row.id, row_title, row_external_id, error=error)
+    except Exception as e:
+        raise RegistrationBulkCreationRowError(row.upload.id, row.id, row_title, row_external_id, error=repr(e))
+    logger.info('Approval required for registration [{}]'.format(registration._id))
```

Comment:
Decided to catch all exceptions just in case since the tasks don't have the roll-back on exception that the API has.